src/Planner/PlannerJoinTree.cpp

```diff
@@ -990,40 +990,42 @@ struct UsingAliasKeyActions
     )
         : left_alias_columns_keys(std::make_shared<ActionsDAG>(left_plan_output_columns))
         , right_alias_columns_keys(std::make_shared<ActionsDAG>(right_plan_output_columns))
-    {}
-
-    void addLeftColumn(QueryTreeNodePtr & node, const ColumnsWithTypeAndName & plan_output_columns, const PlannerContextPtr & planner_context)
     {
-        addColumnImpl(left_alias_columns_keys, node, plan_output_columns, planner_context);
-    }
+        for (const auto * input_node : left_alias_columns_keys->getInputs())
+            left_alias_columns_keys->addOrReplaceInOutputs(*input_node);
 
-    void addRightColumn(QueryTreeNodePtr & node, const ColumnsWithTypeAndName & plan_output_columns, const PlannerContextPtr & planner_context)
-    {
-        addColumnImpl(right_alias_columns_keys, node, plan_output_columns, planner_context);
+        for (const auto * input_node : right_alias_columns_keys->getInputs())
+            right_alias_columns_keys->addOrReplaceInOutputs(*input_node);
```

Comment:
Isn't it enough to use `projectInput()`?

Reply:
`projectInput` means 'pass through unknown columns that are not in a DAG', but inputs should be added to output explicitly.